test/opacity_comparison_funcs.jl

```diff
@@ -67,39 +67,19 @@ function _gray05_H_I_partition_func(T)
     10^interpolator(5040.0 / T)
 end
 
-const gray05_partition_funcs =
-    Dict( "H_I" => _gray05_H_I_partition_func,
-          "H_II" => T -> 1.0,
-          "He_I" => T -> (@assert 2520 ≤ T ≤ 25200, 1.0),
-          "He_II" => T -> (@assert 2520 ≤ T ≤ 25200, 2.0),
-          "He_III" => T -> 1.0)
-
 # compute the ratio of the nₑ to n_H (the number density of all H I and H II)
 # this function could be reused in the future for testing solutions to coupled Saha equations
 function free_electrons_per_Hydrogen_particle(nₑ, T, abundances = SSSynth.solar_abundances,
                                               ionization_energies = SSSynth.ionization_energies,
                                               partition_funcs = SSSynth.partition_funcs)
+    #why doesn't this just use nₑ and nₜ directly?
```

Comment:
We use this function in `H2plus_coefficient`. I believe the difference between the `H2plus_coefficient` and the other absorption coefficients computed in this file is that the number density of Hydrogen isn't fully scaled out from the final result. Therefore we really needed to know the "correct" number density of Hydrogen to reproduce the expected result.

The challenge is that the curve we are trying to reproduce only provides nₑ and the Temperature. Thus to get the correct Hydrogen number density, we need to know the ratio of free electrons to Hydrogen for solar abundances (note that the figure never explicitly stated that they assumed solar abundances, but there are hints that this is the case).

We don't use nₜ, because it wasn't specified in the problem. I suppose that we could refactor this function to compute the number of free electrons per non-electron particle under solar abundance, and then use that to compute the ratio of the number of free electrons per H particle. However, that seems to introduce an unnecessary step.

Let me know if this is clear.

Reply:
This makes sense, thanks.